rpcclient/local.go

```diff
@@ -180,6 +190,15 @@ func (l *Local) BroadcastTxSync(ctx context.Context, tx types.Tx) (*ctypes.Resul
 	}
 	res := <-resCh
 	r := res.GetCheckTx()
+
+	if r.Code == abci.CodeTypeOK {
+		err = l.node.P2P.GossipTx(ctx, tx)
+		if err != nil {
+			l.node.Mempool.RemoveTxByKey(mempool.TxKey(tx), true)
```

Comment:
Oh, I see. This now means that the app said this Tx is valid but no-one will ever see it because gossiping failed (and we out self did not receive it via gossiping but locally)? Is that right?

Reply:
yeah, that's correct